app/routes/individual_response.py

```diff
@@ -27,11 +29,12 @@
 )
 
 
+@login_required
```

Comment:
The `login_required` should be removed from all uses of this blueprint, as having it here means it doesn't need to be on each use of the blueprint.